lib/syskit/roby_app/log_transfer_server/spawn_server.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+require "ftpd"
+require "ipaddr"
+
+require "syskit/roby_app/log_transfer_server/driver"
+
+module Syskit
+    module RobyApp
+        module LogTransferServer
+            # Class responsible for spawning an FTP server for transfering logs
+            class SpawnServer
+                attr_reader :port, :interface
+
+                # tgt_dir must be an absolute path
+                def initialize(
+                    tgt_dir,
+                    user,
+                    password,
+                    certfile_path,
+                    account: "",
+                    interface: "127.0.0.1",
+                    tls: :explicit,
+                    port: 0,
+                    session_timeout: default_session_timeout,
+                    nat_ip: nil,
+                    passive_ports: nil,
+                    debug: false,
+                    log: nil,
+                    verbose: false)
+
+                    @data_dir = tgt_dir
+                    @interface = interface
+                    @tls = tls
+                    @certfile_path = certfile_path
+                    @port = port
+                    @auth_level = "password"
+                    @user = user
+                    @password = password
+                    @account = account
+                    @session_timeout = session_timeout
+                    @nat_ip = nat_ip
+                    @passive_ports = passive_ports
+                    @debug = debug
+                    @log = log
+                    @driver = Driver.new(@user, @password, @account, @data_dir)
+                    @server = Ftpd::FtpServer.new(@driver)
+                    configure_server
```

Comment:
I'm not sure there's a lot of value in storing every attribute as instance variable inside the class. You basically store them here, and the only thing you use them for is to setup the FTP server. Just store what really has to be stored, which from what I see is `@debug`

~~~
driver = Driver.new(user, password, account, data_dir)
server = Ftpd::FtpServer.new(driver)
server.interface = interface
server.port = port
server.tls = tls
server.passive_ports = passive_ports
server.certfile_path = certfile_path
server.auth_level = Ftpd.const_get("AUTH_#{@auth_level.upcase}")
server.session_timeout = session_timeout
server.log = make_log
server.nat_ip = nat_ip
@server = server
@server.start
~~~